realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1302,4 +1411,112 @@ public E findFirst() {
             return null;
         }
     }
+
+    /**
+     * Represents a pending asynchronous Realm query.
+     *
+     * Users are responsible of maintaining a reference to {@code AsyncRequest} in order
+     * to call #cancel in case of a configuration change for example (to avoid memory leak, as the
+     * query will post the result to the caller's thread callback)
+     */
+    public class AsyncRequest {
+        private Future<?> pendingQuery;
+        private volatile boolean isCancelled = false;
+
+        public AsyncRequest(Future<?> pendingQuery) {
+            this.pendingQuery = pendingQuery;
+        }
+
+        /**
+         * Attempts to cancel execution of this query.
```

Comment:
But from a user perspective it will effectively look canceled right? Because neither the success nor error callback will be called after triggering this